server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -249,4 +240,23 @@ public synchronized void shutdownNode() {
         }
     }
 
+    /**
+     * Thread will recycle the node when all active sessions are completed
+     */
+    class NodeRecycleThread extends Thread {
+
+        @Override
+        public void run() {
+            appendMsgToCustomLog("Started NodeRecycleThread to restart the node once all the sessions are completed.");
+            while (getTotalUsed() > 1) {
```

Comment:
Don't we want to wait till getTotalUsed is >0 instead?  I.e let the "last" session timeout?